### PR TITLE
Codenarc

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add Codenarc configuration and fix (/ignore) all warnings.

### CODE REVIEW

File: example/src/jobdsl/demo/templateJobs.groovy [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import com.here.demo.jobdsl.TemplateJobBuilder
 
-[1,2,3].each { index ->
+[1, 2, 3].each { index ->
```
Comment (sschuberth) [inline example/src/jobdsl/demo/templateJobs.groovy]: Use `1.upto(3) {` instead?

File: plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/RestJobManagement.groovy [hunk 2]
```diff
@@ -488,10 +488,7 @@ class RestJobManagement extends AbstractJobManagement implements DeferredJobMana
     }
 
     private void countStatus(String status) {
-        if (!statusCounter[status]) {
-            statusCounter[status] = 0
-        }
-        statusCounter[status] = statusCounter[status] + 1
+        statusCounter[status] = statusCounter[status] ? statusCounter[status] + 1 : 1
```
Comment (sschuberth) [inline plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/RestJobManagement.groovy]: Maybe instead `statusCounter[status] = (statusCounter[status] ?: 0) + 1`?

File: example/src/jobdsl/demo/templateJobs.groovy (hunk 1 above)
Comment (mnonnenmacher) [inline example/src/jobdsl/demo/templateJobs.groovy]: Good idea.

File: plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/RestJobManagement.groovy (hunk 2 above)
Comment (mnonnenmacher) [inline plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/RestJobManagement.groovy]: Thanks, even better.

**Pull request merged**